src/java.base/share/classes/java/lang/System.java

```diff
@@ -2563,17 +2562,9 @@ public Thread currentCarrierThread() {
                 return Thread.currentCarrierThread();
             }
 
-            @ChangesCurrentThread
             public <V> V executeOnCarrierThread(Callable<V> task) throws Exception {
-                Thread thread = Thread.currentThread();
-                if (thread.isVirtual()) {
-                    Thread carrier = Thread.currentCarrierThread();
-                    carrier.setCurrentThread(carrier);
-                    try {
-                        return task.call();
-                    } finally {
-                        carrier.setCurrentThread(thread);
-                    }
+                if (Thread.currentThread() instanceof VirtualThread vthread) {
```

Comment:
Any specific reason to don't use Thread.currentThread().isVirtual()?

Reply:
To use the pattern variable to call `executeOnCarrierThread` on it?

Reply:
ough, missed the last words in the line, thanks